### PR TITLE
Fix node resizing E2E test flake

### PR DESCRIPTION
**Pull request by DangerOnTheRanger**
#### What type of PR is this?

/kind cleanup
/kind flake

#### What this PR does / why we need it:

This PR addresses #100230 by increasing the timeout on `should be able to delete nodes`. I believe there's a timing issue going on in the test that is the source of the flakes. Take a look at https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1421881124996517888 (I've reproduced the relevant bits of the logs here):

```
Aug  1 20:03:32.849: INFO: Cluster has reached the desired number of ready nodes 2
ï¿½[1mSTEPï¿½[0m: waiting 1 minute for the watch in the podGC to catch up, remove any pods scheduled on the now non-existent node and the RC to recreate it
```

Note the timestamp. This message indicates the 1 minute timeout for waiting for the new post-resize pod has started. We then see the error message that results in the test failure:

```
Aug  1 20:04:40.487: FAIL: Unexpected error:
    <*errors.errorString | 0xc0049adb70>: {
        s: "failed to wait for pods responding: pod with UID 4c64f3f9-819d-4824-b96e-25c5ef4566cf is no longer a member of the replica set.  Must have been restarted for some reason. 
```

Now note the pod start times here:

```
Logging pods the kubelet thinks is on node bootstrap-e2e-minion-group-t8bd
Aug  1 20:06:36.620: INFO: my-hostname-delete-node-pjbr8 started at 2021-08-01 20:01:18 +0000 UTC (0+1 container statuses recorded)
Aug  1 20:06:36.620: INFO: 	Container my-hostname-delete-node ready: true, restart count 0

Logging pods the kubelet thinks is on node bootstrap-e2e-minion-group-z5nd
Aug  1 20:06:36.942: INFO: my-hostname-delete-node-2j4jb started at 2021-08-01 20:01:18 +0000 UTC (0+1 container statuses recorded)
Aug  1 20:06:36.942: INFO: 	Container my-hostname-delete-node ready: true, restart count 0
Aug  1 20:06:36.942: INFO: my-hostname-delete-node-txqw5 started at 2021-08-01 20:04:38 +0000 UTC (0+1 container statuses recorded)
Aug  1 20:06:36.942: INFO: 	Container my-hostname-delete-node ready: true, restart count 0
```

The pod started at 20:04:38 started a little after the 1 minute timeout would have expired. We see a similar pattern in another flake, https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-serial/1422247794718019584 (I've reproduced the logs again):

```
Aug  2 19:51:29.758: INFO: Cluster has reached the desired number of ready nodes 2
ï¿½[1mSTEPï¿½[0m: waiting 1 minute for the watch in the podGC to catch up, remove any pods scheduled on the now non-existent node and the RC to recreate it
```

And the failure message:

```
Aug  2 19:52:37.255: FAIL: Unexpected error:
    <*errors.errorString | 0xc0035f0650>: {
        s: "failed to wait for pods responding: pod with UID 749774a9-1f7b-4c10-b8d0-594fd5e4afe7 is no longer a member of the replica set.  Must have been restarted for some reason.
```

And the pod start times (check out the pod started at 19:52:30):

```
Logging pods the kubelet thinks is on node bootstrap-e2e-minion-group-nknw
Aug  2 19:54:30.063: INFO: my-hostname-delete-node-58xt5 started at 2021-08-02 19:52:30 +0000 UTC (0+1 container statuses recorded)
Aug  2 19:54:30.063: INFO: 	Container my-hostname-delete-node ready: true, restart count 0
Aug  2 19:54:30.063: INFO: my-hostname-delete-node-4zhm8 started at 2021-08-02 19:49:14 +0000 UTC (0+1 container statuses recorded)
Aug  2 19:54:30.064: INFO: 	Container my-hostname-delete-node ready: true, restart count 0

Logging pods the kubelet thinks is on node bootstrap-e2e-minion-group-w0m5
Aug  2 19:54:30.375: INFO: my-hostname-delete-node-hxfv9 started at 2021-08-02 19:49:14 +0000 UTC (0+1 container statuses recorded)
Aug  2 19:54:30.375: INFO: 	Container my-hostname-delete-node ready: true, restart count 0
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by DangerOnTheRanger** [comment]
CC'ing @kubernetes/sig-node-bugs on this - is the roughly 1m pod startup time to be expected? I'm wondering if extending the timeout is the right approach to take here.

**Comment by aojea** [comment]
> CC'ing @kubernetes/sig-node-bugs on this - is the roughly 1m pod startup time to be expected? I'm wondering if extending the timeout is the right approach to take here.

can we use one of the WaitForPodReady, e2epod.WaitForPodNameRunningInNamespace, ... functions? 
there are some standard timeouts on the framework already, better to use those constants

**Comment by SergeyKanzhelev** [comment]
>  is the roughly 1m pod startup time to be expected? I'm wondering if extending the timeout is the right approach to take here.

/assign @cynepco3hahue 

for taking a look at performace of pod scheduling.

**Comment by DangerOnTheRanger** [comment]
> > CC'ing @kubernetes/sig-node-bugs on this - is the roughly 1m pod startup time to be expected? I'm wondering if extending the timeout is the right approach to take here.
> 
> can we use one of the WaitForPodReady, e2epod.WaitForPodNameRunningInNamespace, ... functions?
> there are some standard timeouts on the framework already, better to use those constants

The primary concern that I have with potentially using `WaitForPodReady` or something similar, is that the way the test works is by comparing the replicaset after it removes a node to what the replicaset was before. The specific error (which bubbles up from [here](https://github.com/kubernetes/kubernetes/blob/9d38c64a7da5d4c7688ba11334485b8cabd9dc1c/test/e2e/framework/pod/resource.go#L97)) comes from `CheckAllResponses` which is checking pod UIDs (as opposed to names) as well. I think `CheckAllResponses` itself could be hypothetically reworked to be timeout-based, but considering it's used by other code in the test framework I'm worried about the potential impact of such a change.

I think the course of action I like best is to use the value of one of the standard timeouts (`podStartTimeout` looks like the right one) as a parameter to the already-existing call to `time.Sleep`. The value hasn't changed in about [2 years](https://github.com/kubernetes/kubernetes/blame/9d38c64a7da5d4c7688ba11334485b8cabd9dc1c/test/e2e/framework/pod/wait.go#L53), so it should be relatively fine to copy it.

File: test/e2e/cloud/gcp/resize_nodes.go [hunk 1]
```diff
@@ -130,9 +130,9 @@ var _ = SIGDescribe("Nodes [Disruptive]", func() {
 			err = e2enode.WaitForReadyNodes(c, int(originalNodeCount-1), 10*time.Minute)
 			framework.ExpectNoError(err)
 
-			ginkgo.By("waiting 1 minute for the watch in the podGC to catch up, remove any pods scheduled on " +
+			ginkgo.By("waiting 2 minutes for the watch in the podGC to catch up, remove any pods scheduled on " +
 				"the now non-existent node and the RC to recreate it")
-			time.Sleep(time.Minute)
+			time.Sleep(2 * time.Minute)
```
[the remark this answers is not in the document]
Comment (cheftako) [reply test/e2e/cloud/gcp/resize_nodes.go:135]: What about something like podStartShortTimeout https://github.com/kubernetes/kubernetes/blob/00aab6312f7889659e68285efec8b95266025896/test/e2e/framework/timeouts.go#L24?

[the remark this answers is not in the document]
Comment (SergeyKanzhelev) [reply test/e2e/cloud/gcp/resize_nodes.go:135]: @DangerOnTheRanger can you please use @cheftako suggested constant?

[the remark this answers is not in the document]
Comment (DangerOnTheRanger) [reply test/e2e/cloud/gcp/resize_nodes.go]: `PodStartShort` is used now, thanks for the constant suggestion!

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
/assign @cheftako 

@cheftako consider adding it to the milestone. Small risk and will ::fingers_crossed:: likely fix the flake

**Pull request merged**